components/net/fetch/methods.rs

```diff
@@ -156,26 +157,33 @@ pub async fn fetch_with_cors_cache(
 
 /// <https://www.w3.org/TR/CSP/#should-block-request>
 pub fn should_request_be_blocked_by_csp(request: &Request) -> csp::CheckResult {
-    let origin = match &request.origin {
-        Origin::Client => return csp::CheckResult::Allowed,
-        Origin::Origin(origin) => origin,
-    };
-    let csp_request = csp::Request {
-        url: request.url().into_url(),
-        origin: origin.clone().into_url_origin(),
-        redirect_count: request.redirect_count,
-        destination: request.destination,
-        initiator: csp::Initiator::None,
-        nonce: String::new(),
-        integrity_metadata: request.integrity_metadata.clone(),
-        parser_metadata: csp::ParserMetadata::None,
-    };
-    // TODO: Instead of ignoring violations, report them.
-    request
-        .csp_list
-        .as_ref()
-        .map(|c| c.should_request_be_blocked(&csp_request).0)
-        .unwrap_or(csp::CheckResult::Allowed)
+    match &request.policy_container {
+        RequestPolicyContainer::Client => csp::CheckResult::Allowed,
```

Comment:
Can `RequestPolicyContainer` ever be `Client` here? If not, then it would probably be better to assert on that instead of returning `Allowed`.

```rust
let RequestPolicyContainer::PolicyContainer(container) = &request.policy_container else {
    panic!("policy container should not be \"client\"");
}
```

I haven't read the spec (will do that once I have time), but if it doesn't tell us how to handle `"client"` here then it's probably a similar issue as https://github.com/whatwg/fetch/issues/1773.

Reply:
Thanks Simon 👍 I will need to look into this - we should (at least) be evaluating this in step 12 of https://fetch.spec.whatwg.org/#concept-fetch however that's missing atm.

Reply:
After some re-reading, it says in the in the spec for [request](https://fetch.spec.whatwg.org/#concept-request) that:

> "client" is changed to a policy container during fetching...

However there are two paths to [main_fetch](https://fetch.spec.whatwg.org/#concept-main-fetch): via [fetching](https://fetch.spec.whatwg.org/#fetching) as indicated above and in that note in the request spec, and via [redirect-fetch.](https://fetch.spec.whatwg.org/#http-redirect-fetch)

It's the latter case, it's much less clear how request gets it's policy container :/

Reply:
I believe [redirect-fetch](https://fetch.spec.whatwg.org/#http-redirect-fetch) is only called after a [main-fetch](https://fetch.spec.whatwg.org/#concept-main-fetch), so the request should have received a policy container earlier already.

Reply:
> I believe [redirect-fetch](https://fetch.spec.whatwg.org/#http-redirect-fetch) is only called after a [main-fetch](https://fetch.spec.whatwg.org/#concept-main-fetch), so the request should have received a policy container earlier already.

Hmm, but `main_fetch` is the final step of `redirect_fetch` is it not?

Reply:
Right. The redirect fetch algorithm is executed once a response received by a fetch operation has a [redirect-status](https://fetch.spec.whatwg.org/#redirect-status) (and in some other cases, but always in response to an *existing* fetch operation). Internally it involves another main fetch, which reuses the same `fetchParams` as the original fetch, so the policy container is already initialized at this point.

At least that is how I read the spec.